fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java

```diff
@@ -194,4 +199,29 @@ public static List<ColumnDef> buildStatsColumnDef(String tableName) {
             throw new StarRocksPlannerException("Not support stats table " + tableName, ErrorType.INTERNAL_ERROR);
         }
     }
+
+    public static Optional<Double> convertStatisticsToDouble(Type type, String statistic) {
+        if (!type.canStatistic()) {
+            throw new StarRocksPlannerException("Error statistic type : " + type.toSql(), ErrorType.INTERNAL_ERROR);
+        }
+        switch (type.getPrimitiveType()) {
+            case BOOLEAN:
+                if (statistic.equalsIgnoreCase("TRUE")) {
+                    return Optional.of(1D);
+                } else {
+                    return Optional.of(0D);
+                }
+            case DATE:
+                return Optional.of((double) getLongFromDateTime(DateUtils.parseStringWithDefaultHSM(
+                        statistic, DateUtils.DATE_FORMATTER_UNIX)));
+            case DATETIME:
+                return Optional.of((double) getLongFromDateTime(DateUtils.parseStringWithDefaultHSM(
+                        statistic, DateUtils.DATE_TIME_FORMATTER_UNIX)));
+            case CHAR:
+            case VARCHAR:
+                return Optional.empty();
+            default:
+                return Optional.of(Double.parseDouble(statistic));
```

Comment:
Have you tested all the remaining types works fine with `Double.parseDouble`?

Reply:
Yes, complex types do not collect statistics, it has been verified in Analyze